File: Programming Languages/4 languages in 2 styles/OO/Python/Python OO/NonABCFilterer.py
import re

class NonABCFilterer:
    def __init__(self):
        pass

class NonABCFiltererImpl(NonABCFilterer):

    def process(self, file_content_list):
        updated_list = []
        for word in file_content_list:
            updated_list.append(re.sub('[^0-9a-z]+', '', word))
        return updated_list

File: Programming Languages/4 languages in 2 styles/OO/Javascript/Javascript OO/Interfaces/NonABCFilterer.mjs
export class NonABCFilterer {
    process(fileContentList){

    }
}

export class NonABCFiltererImpl extends NonABCFilterer {

    process(fileContentList) {
        for (let i = 0; i < fileContentList.length; i++) {
            fileContentList[i] = fileContentList[i].replace(/[^0-9a-z]/gi, '');
        }
        return fileContentList;
    }
}